### DISCUSSION TITLE
HLS segments (number)

### DISCUSSION THREAD

**Discussion by gAlleb** [savonet/liquidsoap#3410] [Q&A] [answered]
Hi!

Maybe a stupid question :) Recently noticed, that the number of HLS generated .ts segments is less by 1 in `liq2.2.1 opam`. 

E.g. if i have 5 `segments` and 5 `segments_overhead` in my script then I get 9 segments in the folder. 
As i recall in previous version (2.2.0) there were 10 segments (5+5). 

I've  just counted :). Indeed in `liq2.2.0 opam` there are 10 .ts files. In `liq2.2.1 opam` - 9 files. Same script. Have tested on two separate machines: Ubuntu 20.04 and MacOS Ventura.

How come? Is it a bug or an improvement of some sort? Reassurance from the community much needed)

Thank you!

**Comment by toots** [comment, marked as answer]
I think that this is because of the new atomic write support. The current segment only appears when it is finished writing. Before that, the current segment was always visible as it was being written.

**Reply by gAlleb** [reply]
Thanks @toots !
